src/ASM/ASMs2D.C

```diff
@@ -181,12 +181,15 @@ bool ASMs2D::read (std::istream& is)
 }
 
 
-bool ASMs2D::write (std::ostream& os, int) const
+bool ASMs2D::write (std::ostream& os, int basis) const
 {
   if (!surf) return false;
+  if (basis > static_cast<int>(this->getNoBasis())) return false;
+  const Go::SplineSurface* spline = this->getBasis(basis);
+  if (!spline)
+    return false;
```

Comment:
Inconsistent style, return on the same line as the if, or on the next line, L187 vs. L189,190

Reply:
okay, i guess i have to stick to the one-line ifs (i really dislike them as it makes stepping through the code in a debugger less convenient and it makes human parsing of code more painful).

Reply:
I know, I also tend to prefer separate line for the return (unless it is so obvious I will never consider a need for breaking the debugger at that line). I only asked for consistency (at least within the same method, not to enforce 1-line policy)